mesonbuild/linkers.py

```diff
@@ -248,3 +263,359 @@ def unix_args_to_native(cls, args):
+
+    def get_linker_output_args(self, outputname: str) -> List[str]:
+        return ['-o', outputname]
+
+    def get_linker_search_args(self, dirname: str) -> List[str]:
+        return ['-L' + dirname]
+
+
+class GNULikeDynamicLinker(UnixLikeDynamicLinker):
+
+    """Base class for GNU ld, gold, and llvm's lld when acting like GNU ld."""
+
+    _buildtype_args = {
+        'plain': [],
+        'debug': [],
+        'debugoptimized': [],
+        'release': ['-Wl,-O1'],
+        'minsize': [],
+        'custom': [],
+    }
+
+    def get_asneeded_args(self) -> List[str]:
+        return ['-Wl,--as-needed']
+
+    def get_std_shared_module_link_args(self, options) -> List[str]:
+        return ['-shared']
+
+    def get_link_whole_for(self, args: List[str]) -> List[str]:
+        return ['-Wl,--whole-archive'] + args + ['-Wl,--no-whole-archive']
+
+    def get_lto_link_args(self) -> List[str]:
+        return ['-flto']
+
+    def get_no_stdlib_link_args(self) -> List[str]:
+        return ['-nostdlib']
+
+    def get_option_link_args(self, options) -> List[str]:
+        if self.compiler_type.is_windows_compiler:
+            return options['c_winlibs'].value[:]
+        return []
+
+    def get_coverage_link_args(self):
+        # XXX: Is this really a gnuism?
+        return ['--coverage']
+
+
+class GNUBFDDynamicLinker(GNULikeDynamicLinker):
+
+    """Concrete implementation of ld.bfd."""
+
+    def __init__(self, exe: str, compiler_type: 'compilers.CompilerType'):
+        super().__init__('bfd', exe, compiler_type)
+
+
+class GNUGoldDDynamicLinker(GNULikeDynamicLinker):
+
+    """Concrete implementation of ld.gold."""
+
+    def __init__(self, exe: str, compiler_type: 'compilers.CompilerType'):
+        super().__init__('gold', exe, compiler_type)
+
+
+class LLDUnixDynamicLinker(GNULikeDynamicLinker):
+
+    """Concrete implementation of lld (Unix-like)."""
+
+    def __init__(self, exe: str, compiler_type: 'compilers.CompilerType'):
+        super().__init__('lld', exe, compiler_type)
+
+
+class MSVCLikeDynamicLinker(DynamicLinker):
+
+    """Shared base class for MSVC linke.exe and other linkers acting like it."""
+
+    _buildtype_args = {
+        'plain': [],
+        'debug': [],
+        'debugoptimized': [],
+        # The otherwise implicit REF and ICF linker optimisations are
+        # disabled by /DEBUG.  REF implies ICF.
+        'release': ['/OPT:REF'],
+        'minsize': ['/INCREMENTAL:NO', '/OPT:REF'],
+        'custom': [],
+    }
+
+    def get_allow_undefined_link_args(self) -> List[str]:
+        return ['/FORCE:UNRESOLVED']
+
+    def get_linker_always_args(self) -> List[str]:
+        return ['/nologo']
+
+    def get_linker_output_args(self, outputname: str) -> List[str]:
+        return ['/OUT:' + outputname]
+
+    def get_linker_search_args(self, dirname: str) -> List[str]:
+        return ['/LIBPATH:' + dirname]
+
+    def get_linker_output_args(self, outputname: str) -> List[str]:
```

Comment:
__[Flake8]__ [F811] redefinition of unused 'get_linker_output_args' from line 478 ([link](https://sider.review/gh/repos/19784232/pulls/5148#issue-3915087))

<sub>posted by Sider</sub>